src/nft/components/details/detailsV2/RarityGraph.tsx

```diff
@@ -0,0 +1,93 @@
+import Row from 'components/Row'
+import { Trait } from 'nft/types'
+import styled from 'styled-components/macro'
+import { colors } from 'theme/colors'
+
+const RarityBar = styled.div<{ $color?: string }>`
+  background: ${({ $color, theme }) => $color ?? theme.backgroundOutline};
+  width: 2px;
+  height: 10px;
+  border-radius: 2px;
+`
+
+interface RarityValue {
+  threshold: number
+  color: string
+}
+
+enum RarityLevel {
+  VeryCommon = 'Very Common',
+  Common = 'Common',
+  Rare = 'Rare',
+  VeryRare = 'Very Rare',
+  ExtremelyRare = 'Extremely Rare',
+}
+
+const RarityLevels: Map<string, RarityValue> = new Map([
```

Comment:
this might be better as an object to avoid the conditionals

Reply:
The reason I didn't use an object is that with an object I can't use the `RarityLevel` enum as the key for the respective `RarityValues`. Having that is worth the tradeoff of needing the superfluous conditional when getting the threshold

Reply:
can you not not do it like this?

```
const RarityLevels: { [key in RarityLevel]: RarityValue } = {
  [RarityLevel.VeryCommon]: {
    threshold: 0.8,
    color: colors.gray500,
  },
  [RarityLevel.Common]: {
    threshold: 0.6,
    color: colors.green300,
  },
  [RarityLevel.Rare]: {
    threshold: 0.4,
    color: colors.blueVibrant,
  },
  [RarityLevel.VeryRare]: {
    threshold: 0.2,
    color: colors.purpleVibrant,
  },
  [RarityLevel.ExtremelyRare]: {
    threshold: 0,
    color: colors.magentaVibrant,
  },
}

function getRarityLevel(rarity: number) {
  switch (true) {
    case rarity > RarityLevels[RarityLevel.VeryCommon].threshold:
      return RarityLevels[RarityLevel.VeryCommon]
    case rarity > RarityLevels[RarityLevel.Common].threshold:
      return RarityLevels[RarityLevel.Common]
    case rarity > RarityLevels[RarityLevel.Rare].threshold:
      return RarityLevels[RarityLevel.Rare]
    case rarity > RarityLevels[RarityLevel.VeryRare].threshold:
      return RarityLevels[RarityLevel.VeryRare]
    case rarity >= RarityLevels[RarityLevel.ExtremelyRare].threshold:
      return RarityLevels[RarityLevel.ExtremelyRare]
    default:
      return RarityLevels[RarityLevel.VeryCommon]
  }
}
```

Reply:
I stand corrected, great solution